File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/ui/main/MainFragment.java
package com.elveselimoski.starterapp.ui.main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.elveselimoski.starterapp.R;
import com.elveselimoski.starterapp.base.BaseFragment;

public class MainFragment extends BaseFragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void injectFragment() {
        getComponent().inject(this);
    }
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/ui/main/MainFragment.kt
package com.elveselimoski.starterapp.ui.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.elveselimoski.starterapp.R
import com.elveselimoski.starterapp.base.BaseFragment

class MainFragment : BaseFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_main, container, false)
    }

    override fun injectFragment() {
        component.inject(this)
    }
}